### ISSUE TITLE
Class-attribute writes are not resolved through metaclass `__setattr__` (but reads via `__getattr__` are)

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ty#3845]
## Summary

ty does not resolve class-attribute writes through a metaclass `__setattr__`, even though it resolves class-attribute reads through the same metaclass's `__getattr__`, and resolves instance-attribute writes through an instance-level `__setattr__`. Writing an otherwise-undeclared class attribute raises a false `unresolved-attribute`. mypy accepts all three.

## Reproduction

ty 0.0.51, Python 3.10:

```python
# Instance-level __setattr__ -- honored by ty
class WithInstanceSet:
    def __setattr__(self, name: str, value: int) -> None: ...

def f(o: WithInstanceSet) -> None:
    o.anything = 5          # OK

# Metaclass-level __getattr__ / __setattr__
class Meta(type):
    def __getattr__(cls, name: str) -> int: return 0
    def __setattr__(cls, name: str, value: int) -> None: ...

class C(metaclass=Meta): ...

def g(cls: type[C]) -> None:
    _ = cls.anything        # OK   -- read resolved via metaclass __getattr__
    cls.anything = 5        # error -- write NOT resolved via metaclass __setattr__
```

ty output:

```
error[unresolved-attribute] Unresolved attribute `anything` on type `type[C]`.
```

on `cls.anything = 5` only. mypy reports `Success: no issues found`.

## Why it looks like a bug

The three resolution paths are inconsistent:

| access | mechanism | ty |
| --- | --- | --- |
| `o.anything` read/write | instance `__getattr__` / `__setattr__` | honored |
| `cls.anything` read | metaclass `__getattr__` | honored |
| `cls.anything = ...` write | metaclass `__setattr__` | not honored |

Since ty already honors metaclass `__getattr__` for reads and instance `__setattr__` for writes, the natural expectation is that metaclass `__setattr__` is honored for class-attribute writes too.

## Real-world case

This shows up in a schema-DSL whose metaclass defines both hooks, so `db.schema.some_prop = SomeProp(...)` is a supported way to add a property to a schema class. The read side (`db.schema.some_prop`) checks cleanly, but every attribute-style assignment needs a suppression solely because of the metaclass `__setattr__` gap.

## Environment

- ty 0.0.51
- Python 3.10

**Comment by carljm** [comment]
This does look like a valid issue; reopening.
